src/PluginStartupRegistration.h

```diff
@@ -0,0 +1,53 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file PluginStartupRegistration.h
+
+  @author Vitaly Sverchinsky
+
+**********************************************************************/
+
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <wx/string.h>
+#include "AsyncPluginValidator.h"
+
+namespace BasicUI
+{
+   class ProgressDialog;
+}
+
+//Helper class that passes plugins provided in constructor
+//to plugin validator, then "good" plugins are registered in
+//PluginManager. 
+class PluginStartupRegistration final :
+   public AsyncPluginValidator::Delegate
+{
+   std::unique_ptr<BasicUI::ProgressDialog> mDialog;
+   std::unique_ptr<AsyncPluginValidator> mValidator;
+   std::vector<std::pair<wxString, wxString>> mPlugins;
```

Comment:
It's good to make members of new classes `const` where possible as an aid to understanding.  This is one member that could be const.